proposed/2023/Add-dotnet-search.md

```diff
@@ -0,0 +1,100 @@
+Imagine you have your dotnet CLI open. You would like to get the list of all packages from a source `<MySource>`. No worries, you can just use the following command : `dotnet package search --source <MySource>`. It will provide you with the list of all the packages in the source `<MySource>`.
+### Technical explanation
+
+<!-- Explain the proposal in sufficient detail with implementation details, interaction models, and clarification of corner cases. -->
+The `package search [search terms] [options]` command will have the following options 
+
+| Option | Function |
+|---------|:----------|
+| `--Source` | A list of packages to search |
+| `--ExactMatch` | Return exact matches only as a search result |
+| `--Verbose` | Displays a detailed ist of information for each package |
+| `--Prerelease` | Allow prerelease packages to be shown. |
+| `--IncludeDelisted` | Allow unlisted packages to be shown |
+| `--NonInteractive` | Do not prompt for user input or confirmations.|
+| `--Help` | Show command help and usage information |
+| `--Verbosity` | Display the amount of details in the output: normal, quiet, detailed. |
+| `--ConfigFile` | The NuGet configuration file. If not specified, the hierarchy of configuration files from the current directory will be used. |
+|||
+#### **Algorithm**
+1. Read and parse arguments from user
+2. Use the nuget.exe Search API to look for the specific term in the provided source. The search filter will be composed of the `--Source`, `Search term`, `--Prerelease`, `--IncludeDelisted` values.
+3. Based on `--Verbosity` value, provide with a list of packages with various verbosity
+* Verbosity
+    - **Quiet** : Each line would look as follows : 
+    
+            <Package Name> <Latest Package Version>
+    - **Normal** :
+
+                <Package Name> | <Latest Package Version> | <Amount of Downloads>
+                <Description of the Package(short form)>
+    - **Detailed** :
+
+                <Package Name> | <Latest Package Version> | <Amount of Downloads>
+                Deprecated : YES/NO | Vulnerable : YES/NO
+                <Description of the Package(Readme file)>
+                License URL : <URL>
+
+#### **New Option `ExactMatch`**
+* This new option will allow for users to be able to search and have only exact matches as an output. 
+* For example if a user uses `dotnet package search NuGet.CommandLine --Verbosity Quiet`
+
+        NuGet.CommandLine | 6.7.0
+        NuGet.CommandLine.XPlat | 6.7.0
+        NuGet.Commands | 6.7.0
+        NuGet.exe | 3.4.3
+        NuGet.Bootstrapper | 2.6.0
+        CommandLineParser20 | 2.0.0
+        NuGet.VerifyMicrosoftPackage | 1.0.0
+        NuGet.for.MSBuild | 2.1.0 ...
+* Using ``dotnet package search NuGet.CommandLine --ExactMatch --Verbosity Quiet`` on the other side will have the following output
+
+         NuGet.CommandLine | 6.7.0
+## Drawbacks
+
+<!-- Why should we not do this? -->
```

Comment:
It hammers the server, which is a reason not to do it. Consider nuget.org, which claims to have 371,611 unique package IDs. What's the customer scenario where they "need" to list over 370,000 packages on the server?

Additionally, the request to add a `nuget list` equivalent to the `dotnet` CLI was created before a `dotnet nuget search` command existed. Therefore, it's entirely possible that a `dotnet nuget search` makes `nuget list` unnecessary.

Reply:
I'm guessing this is meant to be resolved.